mesh/mesh.go

```diff
@@ -667,10 +667,12 @@ func (msh *Mesh) AddBlockWithTXs(ctx context.Context, block *types.Block) error
 	msh.setLatestLayer(logger, block.LayerIndex)
 	logger.Debug("associated txs to block")
 
+	// add block to the tortoise before storing it
+	// otherwise fetcher will not wait until data is stored in the tortoise
```

Comment:
the comment `fetcher will not wait` isn't clear to me. can you elaborate a little bit more please? thanks

Reply:
the assumption was that fetcher returns without error either:
- block is already available in the local database
- after block is downloaded from peer and passes handler without error

if block is saved before it is added to the database there is a race condition and tortoise.DecodeBallot can be called after block was saved to db but before OnBlock was called